tests/contracts/tx_vec/src/lib.rs

```diff
@@ -103,7 +103,7 @@ mod hosted {
 
                 let mut sink = Sink::new(&mut bytes[..]);
                 // return new state
-                &ContractState::from_canon(&slf).encode(&mut sink);
+                let _ = &ContractState::from_canon(&slf).encode(&mut sink);
```

Comment:
Handle the error or add a comment why it's not possible

Reply:
Notice that is not part of this PR, it was just added due the clippy warnings. I can remove it and leave out of this PR if it make this a blocker, and we can have a follow up PR where we take in account all the tests issue.